src/main/kotlin/uk/gov/justice/hmpps/probationsearch/contactsearch/ContactSearchController.kt

```diff
@@ -31,26 +39,50 @@ class ContactSearchController(
   ): ContactSearchResponse {
     val useSemanticSearch = semantic ?: featureFlags.enabled(FeatureFlags.SEMANTIC_CONTACT_SEARCH)
     return if (useSemanticSearch) {
-      val started = Instant.now()
-      val response = contactSearchService.semanticSearch(request, pageable)
-      telemetryClient.trackEvent(
-        "SemanticSearchCompleted",
-        mapOf(
-          "crn" to request.crn,
-          "query" to request.query.length.toString(),
-          "resultCount" to response.totalResults.toString(),
-          "queryTermCount" to TermSplitter.split(request.query).size.toString(),
-          "page" to pageable.pageNumber.toString(),
-          "resultCountForPage" to response.results.size.toString(),
-          "semanticOnlyResultCountForPage" to response.results.count { it.highlights.isEmpty() }.toString()
-        ),
-        mapOf(
-          "duration" to started.until(Instant.now(), ChronoUnit.MILLIS).toDouble(),
-        ),
-      )
+      val (response, duration) = measureTimedValue { contactSearchService.semanticSearch(request, pageable) }
+      trackSemanticSearch(request, response, pageable, duration)
       response
     } else {
+      val auth = SecurityContextHolder.getContext().authentication
+      GlobalScope.launch(Job() + Context.current().asContextElement()) {
+        SecurityContextHolder.getContext().authentication = auth
```

Comment:
Is there any risk of mixing authentication between threads here, given we're using a global scope?  Seems worrying that even the Kotlin docs refer to it as a "delicate" API... https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-global-scope/

Reply:
I suppose it doesn't matter too much, thinking about it, as there's only a single set of auth credentials for the client - still feels a bit iffy though.

We should at least timeout the job either way after say 30 seconds, to ensure they don't accidentally run forever.

Also might be worth increasing the replica count to account for any extra load while this runs, just in case, I set it to 8 last time, here: https://github.com/ministryofjustice/probation-offender-search/pull/796/files#diff-bb4e80ba0a6c43663498b44c19e42c074dcceac3d9564b49801202073b5c87ffR5